src/tree/root.ts

```diff
@@ -10,7 +10,7 @@ import {
 export type Data = {
   tree: Tree | null;
   breakpoints: Array<Breakpoint>;
-  props: Array<InstanceProps>;
+  props: Array<InstanceProps & { props: Props[] }>;
```

Comment:
I think I made it all confusing with props and instanceProps, maybe you have an idea for the future iteration how to name them